### PR TITLE
SOV-4139: BoB Spice leaderboard

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-4139

### CODE REVIEW

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/components/BalanceRenderer/BalanceRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+type BalanceRendererProps = {
+  user: User;
+};
+
+export const BalanceRenderer: FC<BalanceRendererProps> = ({ user }) => {
+  const { account } = useAccount();
+  const usersDepositsData: UserDeposit[] = usersDeposits;
+  const deposits = usersDepositsData.filter(deposit =>
+    areAddressesEqual(deposit.userAddress, user.wallet),
+  );
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/components/BalanceRenderer/BalanceRenderer.tsx]: put it under useMemo

**Review by creed-victor — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/LeaderboardPointsPage.constants.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export const COLUMNS_CONFIG = (isUserPosition: boolean = false) => [
+  {
+    id: '',
+    title: '',
+    cellRenderer: (user: User) => user.id,
+    className: 'hidden lg:table-cell',
+  },
+  {
+    id: 'position',
+    title: isUserPosition
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/LeaderboardPointsPage.constants.tsx]: This doesn't make sense.
image

Take a look at how it's handled in the POWA page and do it similarly (consult with fineline or Hyde if not sure): https://github.com/DistributedCollective/sovryn-dapp/blob/98b527d4add1eb75e1f9db7da6dc534c86e70985/apps/frontend/src/app/5_pages/LeaderboardPage/components/Leaderboard/components/BaseTable/BaseTable.constants.tsx#L22

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/LeaderboardPointsPage.constants.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      : t(translations.leaderboardPointsPage.table.position),
+    cellRenderer: (user: User) => (
+      <div className="text-right lg:text-left w-full">
+        {prettyTx(user.wallet, 4)}
+      </div>
+    ),
+    className: 'hidden lg:table-cell',
+  },
+  {
+    id: 'balance',
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/LeaderboardPointsPage.constants.tsx:38]: Consult with fineline and Hyde but I don't like the fact that `Balance` is shown only for the connected wallet and it's empty for all the rows.

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import {
+  EXTRA_SPICE_POINTS_MULTIPLIER,
+  RUNES_POINTS_MULTIPLIER,
+} from '../LeaderboardPointsPage.constants';
+import { User, UserPoints } from '../LeaderboardPointsPage.types';
+import usersPointsList from '../data/usersPoints.json';
+
+export const useGetPoints = (pageSize: number, page: number) => {
+  const { account } = useAccount();
+  const usersPointsData: UserPoints[] = usersPointsList;
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts]:
```suggestion
  const data: UserPoints[] = usersPointsList;
```
Move it above the hook.

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  const startIndex = page * pageSize;
+  const endIndex = Math.min(startIndex + pageSize, usersPointsList.length);
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts]:
```suggestion
  const startIndex = page * pageSize;
  const endIndex = Math.min(startIndex + pageSize, usersPointsList.length);
```
Use `useMemo`.

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    return usersPointsData.sort((a, b) => b.points - a.points);
+  }, [usersPointsData]);
+
+  const userIndex = useMemo(() => {
+    return sortedUsers.findIndex(user =>
+      areAddressesEqual(user.wallet, account),
+    );
+  }, [sortedUsers, account]);
+
+  const userPoints: User[] = useMemo(() => {
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts]:
```suggestion
  const connectedWalletPoints: User[] = useMemo(() => {
```

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          spice: user.points,
+          extraSpice: user.points * EXTRA_SPICE_POINTS_MULTIPLIER,
+          runes: user.points * RUNES_POINTS_MULTIPLIER,
+        },
+      ];
+    }
+    return [];
+  }, [sortedUsers, userIndex]);
+
+  const usersPoints: User[] = useMemo(
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts]:
```suggestion
  const points: User[] = useMemo(
```

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+      const user = sortedUsers[userIndex];
+      return [
+        {
+          id: userIndex + 1,
+          wallet: user.wallet,
+          spice: user.points,
+          extraSpice: user.points * EXTRA_SPICE_POINTS_MULTIPLIER,
+          runes: user.points * RUNES_POINTS_MULTIPLIER,
+        },
+      ];
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts]:
```suggestion
      const {wallet, points} = sortedUsers[userIndex];
      return [
        {
          id: userIndex + 1,
          wallet,
          spice: points,
          extraSpice: points * EXTRA_SPICE_POINTS_MULTIPLIER,
          runes: points * RUNES_POINTS_MULTIPLIER,
        },
      ];
```

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+  const usersPoints: User[] = useMemo(
+    () =>
+      sortedUsers.slice(startIndex, endIndex).map((user, index) => ({
+        id: startIndex + index + 1,
+        wallet: user.wallet,
+        spice: user.points,
+        extraSpice: user.points * EXTRA_SPICE_POINTS_MULTIPLIER,
+        runes: user.points * RUNES_POINTS_MULTIPLIER,
+      })),
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/hooks/useGetPoints.ts]:
```suggestion
      sortedUsers.slice(startIndex, endIndex).map(({wallet, points}, index) => ({
        id: startIndex + index + 1,
        wallet,
        spice: points,
        extraSpice: points * EXTRA_SPICE_POINTS_MULTIPLIER,
        runes: points * RUNES_POINTS_MULTIPLIER,
      })),
```

**Review by tiltom — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/LeaderboardPointsPage/components/ConnectWalletMessage/ConnectWalletMessage.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+import { t } from 'i18next';
+
+import { Button } from '@sovryn/ui';
+
+import { useWalletConnect } from '../../../../../hooks';
+import { translations } from '../../../../../locales/i18n';
+
+export const ConnectWalletMessage = () => {
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/LeaderboardPointsPage/components/ConnectWalletMessage/ConnectWalletMessage.tsx]: Can you just reuse the existing component and move it to `app/1_atoms` or `app/2_molecules`?

**Review by tiltom — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**